packages/amplify_datastore/example/ios/unit_tests/PublisherExtensionTests.swift

```diff
@@ -0,0 +1,109 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+
+import XCTest
+import Combine
+@testable import amplify_datastore
+
+class PublisherExtensionTests: XCTestCase {
+    func testToAmplifyAsyncThrowingSequence_createAsyncSequenceCorrect() async throws {
+        let subject = PassthroughSubject<Int, TestError>()
+        let (sequence, cancellable) = subject.eraseToAnyPublisher().toAmplifyAsyncThrowingSequence()
+
+        let expectation1 = expectation(description: "element 1 received")
+        let expectation2 = expectation(description: "element 2 received")
+        let expectation3 = expectation(description: "element 3 received")
+        Task {
+            for try await element in sequence {
+                switch element {
+                case 1: expectation1.fulfill()
+                case 2: expectation2.fulfill()
+                case 3: expectation3.fulfill()
+                default: break
+                }
+            }
+        }
+
+        Task {
+            subject.send(1)
+            subject.send(2)
+            subject.send(3)
+            subject.send(completion: .finished)
+        }
+
+        await fulfillment(of: [expectation1, expectation2, expectation3], timeout: 1)
```

Comment:
The tests are failing in CI with the message 

```
❌  /Users/runner/work/amplify-flutter/amplify-flutter/packages/amplify_datastore/example/ios/unit_tests/PublisherExtensionTests.swift:39:15: cannot find 'fulfillment' in scope
```

Reply:
Thanks for reviewing @Jordan-Nelson.

The API `fulfillment` mentioned in the error log was introduced in Xcode 14.3.

As discussed with @Equartey, to utilize the datastore v2, flutter will need to update the CI GitHub runner to `macOS-14` or `macOS-14-arm64`, as the minimum supported version for datastore v2 has been updated to Swift 5.9.